### DISCUSSION TITLE
CoreDNS and SMB mount name resolution

### DISCUSSION THREAD

**Discussion by rogerfn** [kubernetes-csi/csi-driver-smb#499] [Q&A]
Hi everyone,

I'm not sure if this is a bug or something, I need a bit of help here.

I have a custom dns server configuration in CoreDNS that apparently is ignored by the SMB mount.  The Pod when it tries to mount the share, cannot resolve the host name xxxx.mycompany.com. All the pods in the cluster can resolve xxxx.mycompany.com. The connection works fine if I use the IP address instead

```
-------
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns-custom
  namespace: kube-system
data:
  companydns.server: |
    
    mycompany.com:53 {
      errors
      cache 1
      forward . 10.10.1.1 10.10.1.2

    }
```

**Comment by andyzhangx** [comment]
what's the `dnsPolicy` of the csi-smb-node daemonset? 
https://github.com/kubernetes-csi/csi-driver-smb/blob/2457c6d71e4b03e811b7a006baa3c08beee8d02c/deploy/csi-smb-node.yaml#L21

And also have you tried restart the coredns daemonset pods?

**Reply by rogerfn** [reply]
I'm using the Default one. Yes, I tried to restart the CoreDNS daemonset pods. The pod can resolve the name when I don't add the volume. 

Cluster Info: AKS. Private Cluster, Kubelet, Calico.

**Reply by rogerfn** [reply]
ClusterFirstWithHostNet did the trick actually, thanks for pointing this out

**Reply by andyzhangx** [reply]
`ClusterFirstWithHostNet` would fix the issue?

**Reply by rogerfn** [reply]
Yes, not sure why to be honest. I'm running kubenet and my vnet is using Azure-provided DNS. I had to add a custom CoreDNS to be able to resolve the internal hostnames.   My pod doesn't specify hostNetwork.

**Reply by andyzhangx** [reply]
there are also cases there `ClusterFirstWithHostNet` would break for on-prem smb server name resolution: https://github.com/kubernetes-csi/csi-driver-smb/pull/422

**Reply by rogerfn** [reply]
After few days, I restarted it and it broke and it can't resolve the name anymore with none of the dnsPolicy. I did notice that all the resolv.conf of all the pods points to the kubernetes DNS server ... but the smb node pod points to the azure one 168.63.129.16, ignoring the policy that is passed

**Comment by rogerfn** [comment]
I used dnsPolicy: None and then manually configured the DNS.. Something like:

```yaml
dnsPolicy: None 
dnsConfig:
  nameservers: ["10.138.16.10"] # My DNS Service IP
  searches:
  - cluster.local
  - svc.cluster.local
  options:
  - name: ndots
    value: "5"
```

I think it will be quite good if we have that posibility in the Charts
